script/startnew.py

```diff
@@ -33,10 +34,34 @@
 def main():
     print EXPLANATION
 
+    # Get the values.
+    try:
+        while True:
+            short_name = raw_input("Short name: ")
+            if re.match(r"^[a-z][a-z0-9_]+$", short_name):
+                break
+            print "The short name must be a valid Python identifier, all lower-case."
```

Comment:
I thought about that, but I would rather have the user pick a name that is lower-case.  I might make the name different if I realized ahead of time that it would be squashed down.